### PR TITLE
[Merged by Bors] - Use only lighthouse types in the mock builder

### PR DESCRIPTION
**Pull request by realbigsean**
## Proposed Changes

- only use LH types to avoid build issues
- use warp instead of axum for the server to avoid importing the dep

## Additional Info

- wondering if we can move the `execution_layer/test_utils` to its own crate and import it as a dev dependency
- this would be made easier by separating out our engine API types into their own crate so we can use them in the test crate
- or maybe we can look into using reth types for the engine api if they are in their own crate

### CODE REVIEW

File: beacon_node/execution_layer/src/test_utils/mock_builder.rs [hunk 1]
```diff
@@ -58,119 +37,128 @@ pub enum Operation {
...
 
-impl BidStuff for BuilderBid {
-    fn fee_recipient_mut(&mut self) -> &mut ExecutionAddress {
-        match self {
-            Self::Bellatrix(bid) => &mut bid.header.fee_recipient,
-            Self::Capella(bid) => &mut bid.header.fee_recipient,
+impl<E: EthSpec> BidStuff<E> for BuilderBid<E, BlindedPayload<E>> {
+    fn set_fee_recipient(&mut self, fee_recipient: Address) {
+        match self.header.to_mut() {
+            BlindedPayloadRefMut::Merge(payload) => {
```
Comment (michaelsproul) [inline beacon_node/execution_layer/src/test_utils/mock_builder.rs:78]: Wanted to rewrite this with a mapping macro but realised it isn't possible. Opened an issue on superstruct: https://github.com/sigp/superstruct/issues/31

**Review by michaelsproul — APPROVED** [review]
Fantastic! Should hopefully make compilation a bit lighter with those deps removed

File: beacon_node/beacon_chain/src/test_utils.rs [hunk 2]
```diff
@@ -595,7 +595,10 @@ where
             .execution_block_generator()
     }
 
-    pub fn set_mock_builder(&mut self, beacon_url: SensitiveUrl) -> MockBuilderServer {
+    pub fn set_mock_builder(
+        &mut self,
+        beacon_url: SensitiveUrl,
+    ) -> impl std::future::Future<Output = ()> {
```
Comment (jxs) [inline beacon_node/beacon_chain/src/test_utils.rs]: nit:
```suggestion
    pub async fn set_mock_builder(&mut self, beacon_url: SensitiveUrl)  {
```
and bellow call ` mock_builder_server.await`

File: beacon_node/execution_layer/src/test_utils/mock_builder.rs [hunk 3]
```diff
@@ -193,7 +180,7 @@ impl<E: EthSpec> MockBuilder<E> {
         beacon_url: SensitiveUrl,
         spec: ChainSpec,
         executor: TaskExecutor,
-    ) -> (Self, MockBuilderServer) {
+    ) -> (Self, (SocketAddr, impl Future<Output = ()>)) {
```
Comment (jxs) [inline beacon_node/execution_layer/src/test_utils/mock_builder.rs:183]: same as above, we can just:
```suggestion
    ) -> (Self, SocketAddr) {
```
if we change the signature of the function to `async fn` and `.await` on `server`

**Review by jxs — COMMENTED** [review]
Thanks Sean!

File: beacon_node/execution_layer/src/test_utils/mock_builder.rs (hunk 3 above)
Comment (michaelsproul) [reply beacon_node/execution_layer/src/test_utils/mock_builder.rs:183]: We want to lazily evaluate the `server` future though, after the socket address has been returned and used by the caller.

File: beacon_node/beacon_chain/src/test_utils.rs (hunk 2 above)
Comment (realbigsean) [reply beacon_node/beacon_chain/src/test_utils.rs]: true! updated

File: beacon_node/execution_layer/src/test_utils/mock_builder.rs (hunk 3 above)
Comment (realbigsean) [reply beacon_node/execution_layer/src/test_utils/mock_builder.rs:183]: Oh yea, this one I think we have to leave

File: beacon_node/beacon_chain/src/test_utils.rs (hunk 2 above)
Comment (realbigsean) [reply beacon_node/beacon_chain/src/test_utils.rs]: `impl Future` actually implicitly has a `'static` lifetime, and switching to `async` changes this, so had to change it back

**Comment by realbigsean** [comment]
bors retry

**Comment by realbigsean** [comment]
bors retry

**Comment by realbigsean** [comment]
bors retry

**Comment by realbigsean** [comment]
bors retry

**Pull request closed without merging**